pdns/recursor_cache.cc

```diff
@@ -240,7 +240,7 @@ void MemRecursorCache::replace(time_t now, const DNSName &qname, const QType& qt
 {
   d_cachecachevalid = false;
   //  cerr<<"Replacing "<<qname<<" for "<< (ednsmask ? ednsmask->toString() : "everyone") << endl;
-  auto key = boost::make_tuple(qname, qt.getCode(), ednsmask ? *ednsmask : Netmask());
+  auto key = boost::make_tuple(qname, qt.getCode(), ednsmask ? (*ednsmask).getNormalized() : Netmask());
```

Comment:
This change only works as long as the `NetmaskTree` keeps normalizing the keys, I'd prefer we make sure that the key we insert into the ECS index is always the same than the one we insert into the 'regular' index instead.

```suggestion
  if (ednsmask) {
    ednsmask = ednsmask->getNormalized();
  }
  auto key = boost::make_tuple(qname, qt.getCode(), ednsmask ? *ednsmask : Netmask());
```

Reply:
changed